### PR TITLE
Add explicit security context capability for compatibility with cri-o

### LINKED ISSUE #620: chroot not permitted with cri-o
Attempting to deploy CrateDB with the operator with https://cri-o.io/ results in 
```
chroot: cannot change root directory to '/': Operation not permitted
```

Steps to reproduce:

```bash
sudo snap install microk8s --classic --channel=1.29/stable
export OS=xUbuntu_22.04
export CRIO_VERSION=1.22
echo "deb https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/$OS/ /"| sudo tee /etc/apt/sources.list.d/devel:kubic:libcontainers:stable.list
echo "deb http://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable:/cri-o:/$CRIO_VERSION/$OS/ /"|sudo tee /etc/apt/sources.list.d/devel:kubic:libcontainers:stable:cri-o:$CRIO_VERSION.list
curl -L https://download.opensuse.org/repositories/devel:kubic:libcontainers:stable:cri-o:$CRIO_VERSION/$OS/Release.key | sudo apt-key add -
curl -L https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/$OS/Release.key | sudo apt-key add -
sudo apt update
sudo apt install cri-o cri-o-runc -y
sudo systemctl start crio
sudo systemctl enable crio
sudo apt install -y containernetworking-plugins
sudo systemctl restart crio
sudo apt install -y cri-tools
sudo crictl --runtime-endpoint unix:///var/run/crio/crio.sock version
sudo nano /var/snap/microk8s/current/args/kubelet
```
Replace the `container-runtime-endpoint` line with:
```
--container-runtime-endpoint=unix:///var/run/crio/crio.sock
--runtime-request-timeout=10m
--cgroup-driver="systemd"
```

On the steps below replace `hernan` with your user name:
```bash
sudo microk8s stop
sudo microk8s start
sudo usermod -a -G microk8s hernan
sudo chown -R hernan ~/.kube
newgrp microk8s

microk8s kubectl create namespace crate-operator
microk8s kubectl create namespace crate

nano StorageClassMicrok8s.yaml
```

```yaml
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: default
provisioner: microk8s.io/hostpath
reclaimPolicy: Delete
volumeBindingMode: Immediate
```

```bash
microk8s enable storage
microk8s kubectl apply -f StorageClassMicrok8s.yaml

microk8s helm repo add crate-operator https://crate.github.io/crate-operator
microk8s helm repo update
microk8s helm install crate-operator crate-operator/crate-operator --namespace crate-operator

nano dev-cluster.yaml
```

```yaml
apiVersion: cloud.crate.io/v1
kind: CrateDB
metadata:
  name: my-cluster
  namespace: crate
spec:
  cluster:
    imageRegistry: crate
    name: crate-dev
    version: 5.7.1
  nodes:
    data:
    - name: my-cluster
      replicas: 1
      resources:
        limits:
          cpu: 1
          memory: 1024Mi
        disk:
          count: 1
          size: 2GiB
          storageClass: default
        heapRatio: 0.25
```

```bash
microk8s kubectl --namespace crate create -f dev-cluster.yaml
microk8s kubectl get cratedbs -n crate
microk8s kubectl get pods --namespace crate 
microk8s kubectl logs crate-data-my-cluster-my-cluster-0 crate -n crate
```

```
chroot: cannot change root directory to '/': Operation not permitted
```

```bash
sudo nano /etc/crio/crio.conf.d/01-crio-runc.conf
```

```
default_capabilities = ["SYS_CHROOT"]
```

```bash
sudo systemctl restart crio
```
--> same issue

### PR DESCRIPTION
**Pull request by hlcianfagna**
## Summary of changes

Fixes #620

## Checklist

- [X] Link to issue this PR refers to:
- [X] Relevant changes are reflected in `CHANGES.rst`
- [ ] Added or changed code is covered by tests
- [ ] Documentation has been updated if necessary
- [ ] Changed code does not contain any breaking changes (or this is a major version change)

### CODE REVIEW

File: crate/operator/create.py [hunk 1]
```diff
@@ -611,7 +612,7 @@ def get_statefulset_init_containers(crate_image: str) -> List[V1Container]:
             image="busybox:1.35.0",
             image_pull_policy="IfNotPresent",
             name="init-sysctl",
-            security_context=V1SecurityContext(privileged=True),
+            security_context=V1SecurityContext(capabilities=V1Capabilities(add=["SYS_CHROOT"])),
```
Comment (WalBeh) [inline crate/operator/create.py:615]:
```suggestion
            security_context=V1SecurityContext(capabilities=V1Capabilities(add=["SYS_ADMIN"])),
```

While I think it is an improvement, I think this is not the place where we need `SYS_CHROOT`

Comment (hlcianfagna) [reply crate/operator/create.py:615]: Thank you, is there any particular operation that comes to mind for which we need `SYS_ADMIN` ?
Do you think the CI tests in this repo are comprehensive enough or you think a change here would need to be tested in some other way?

Comment (hlcianfagna) [reply crate/operator/create.py:615]: Discussed with @WalBeh , this container should be fine with `SYS_ADMIN` as it runs `sysctl`, it is the CrateDB container that needs `SYS_CHROOT`

File: crate/operator/create.py [hunk 2]
```diff
@@ -618,7 +622,9 @@ def get_statefulset_init_containers(crate_image: str) -> List[V1Container]:
             image="busybox:1.35.0",
             image_pull_policy="IfNotPresent",
             name="init-sysctl",
-            security_context=V1SecurityContext(privileged=True),
+            security_context=V1SecurityContext(
+                capabilities=V1Capabilities(add=["SYS_ADMIN"])
```
Comment (WalBeh) [inline crate/operator/create.py:626]: I had double checked on `SYS_ADMIN`, seems like it does not cut it here, looks like we need to specify:
`security_context=V1SecurityContext(privileged=True),`

**Review by juanpardo — APPROVED** [review]

**Pull request merged**